pkg/utils/common.go

```diff
@@ -30,3 +32,17 @@ func GetNamespacedName(obj client.Object) types.NamespacedName {
 func IsDeleted(obj client.Object) bool {
 	return obj != nil && !obj.GetDeletionTimestamp().IsZero()
 }
+
+// GetVersionFromImage return the version from spec.global.image.
+func GetVersionFromImage(image string) string {
+	version := ""
+	versions := strings.Split(image, ":")
+	if image == "" {
```

Comment:
We can save some computation by bringing this branch to the head of the function.